code/datums/helper_datums/map_template.dm

```diff
@@ -0,0 +1,53 @@
+/datum/map_template
+	var/name = "Default Template Name"
+	var/width = 0
+	var/height = 0
+	var/mappath = null
+	var/mapfile = null
+
+/datum/map_template/New(path=null, map=null, rename=null)
+	if(path)
+		mappath = path
+		preload_size(mappath)
+	if(map)
+		mapfile = map
+	if(rename)
+		name = rename
+
+/datum/map_template/proc/preload_size(path)
+	var/quote = ascii2text(34)
+	var/map_file = file2text(path)
+	var/key_len = length(copytext(map_file,2,findtext(map_file,quote,2,0)))
+	//assuming one map per file since more makes no sense for templates anyway
+	var/mapstart = findtext(map_file,"\n(1,1,") //todo replace with something saner
+	var/content = copytext(map_file,findtext(map_file,quote+"\n",mapstart,0)+2,findtext(map_file,"\n"+quote,mapstart,0)+1)
+	var/line_len = length(copytext(content,1,findtext(content,"\n",2,0)))
+
+	width = line_len/key_len
+	height = length(content)/(line_len+1)
+
+/datum/map_template/proc/load(turf/T)
+	if(!T)
+		return
+	if(T.x+width > world.maxx)
+		return
+	if(T.y+height > world.maxy)
+		return
+
+	maploader.load_map(get_file(), T.x, T.y, T.z)
+	for(var/turf/B in block(T,locate(T.x+width,T.y+height,T.z)))
+		for(var/atom/movable/AM in B)
+			AM.initialize()
```

Comment:
Sure, will change that. I think no turfs actually use initialize but i guess futureproofing this is better.